app/processors/email_processor.rb

```diff
@@ -1,12 +1,25 @@
 class EmailProcessor
   cattr_accessor :default_to_newest_feed, default: false
+  attr_writer :feed
 
-  def initialize(email)
+  def self.for_payload(payload, feed: nil)
+    json = JSON.parse(payload)
+    params = Griddler::Postmark::Adapter.normalize_params(json.deep_symbolize_keys)
+    email = Griddler::Email.new(params)
+    new(email, feed: feed)
+  end
+
+  def initialize(email, feed: nil)
     @email = email
+    @feed = feed
```

Comment:
Isn't this assignment redundant with the `@feed ||= …` side effect in `def feed` below?

I guess `feed` could be changed to:

``` ruby
  def feed
    Feed.find_by(token: feed_token)
  end
```

Reply:
the @feed ||= is actually the only thing that makes this work! if @feed is nil, it will call find_by, but if @feed is assigned to something other than nil (which is the default value), then it won't find a new feed at all

Reply:
oh yes I was confused.

though related to my other comment in `feeds_spec.rb`, I think we could live without the extra parameters in `initialize` and `self.for_payload` for `feed` as well as the `cattr_accessor` and `attr_writer` 😬